packages/components/src/modal/index.tsx

```diff
@@ -350,7 +351,7 @@ function UnforwardedModal(
 	);
 
 	return createPortal(
-		<ModalContext.Provider value={ nestedDismissers.current }>
```

Comment:
I suppose avoiding this reading of the ref during render was the idea behind using state instead. Or was it something else?

Reply:
The state is also often used for one-time initializations. See https://tkdodo.eu/blog/use-state-for-one-time-initializations.

Reply:
The main advantage of `useState` for one-time initialization like this is that it supports an initialization function. You can write:
```js
useState( () => new Set() );
```
and avoid creating a new set on each render. It's created only when `useState` really wants to init and calls the initializer. On the other hand, with `useRef` you can only do this:
```js
useRef( new Set() );
```
In this particular case it doesn't matter much, but for more expensive initializations it can make a difference.

Reply:
Ah, makes sense. Indeed it would hardly matter here and probably neither would there be a consequence to reading the ref in this case. All around it’s better though. Thanks!

Reply:
The `useState` initialization pattern has been becoming popular recently because of React 19 and the React Compiler: I don't remember the exact details right now, but the compiler issues warnings about some `useRef` usages, and switching to `useState` makes it happy.